app/main.py

```
@@ -96,10 +145,243 @@ def _tc_get_json(path, params=None, timeout=30):
+    logging.info("Starting failed-builds (meta-runner) update thread")
+    while True:
+        try:
+            meta_runner_ids = resolve_meta_runner_ids()
+            if not meta_runner_ids:
+                logging.warning(
+                    "No meta-runner ids (discovery empty and META_RUNNER_IDS unset); "
+                    "skipping failed-builds update"
+                )
+            else:
+                update_failed_build_metrics(meta_runner_ids)
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)